stdlib/pervasives.ml

```diff
@@ -369,7 +373,7 @@ external seek_out : out_channel -> int -> unit = "caml_ml_seek_out"
 external pos_out : out_channel -> int = "caml_ml_pos_out"
 external out_channel_length : out_channel -> int = "caml_ml_channel_size"
 external close_out_channel : out_channel -> unit = "caml_ml_close_channel"
-let close_out oc = flush oc; close_out_channel oc
+let close_out oc = (try flush oc with _ -> ()); close_out_channel oc
```

Comment:
@hcarty Ok, I have looked at the documentation and it actually already specifies that `close_out` may raise even while flushing and before closing.  Users who don't like that behavior should use `close_out_noerr`, which completely ignores errors.  This means that the previous implementation in `threads` was not compliant with the manual while the `stdlib` was.

I have just submitted a patch to make the current `stdlib` behavior the default with both implementations.  They must definitely behave the same for that function.  It would be very confusing for users if that wasn't the case.  If some thread-related library needed the previous closing behavior, that code would have to be changed to use `close_out_noerr`.  I have looked, but haven't seen any such code in the distribution.  The semantics as specified in the manual should definitely prevail.